bp_common/src/include/bp_common_me_if.vh

```diff
@@ -183,105 +181,123 @@ typedef enum bit [1:0]
- * addr specifies the memory address associated with the command
- * way_id is the way within the set that addr maps to in the LCE that should be used for the command
- * state specifies the Coherence State to be used for invalidate, set_tag, and set_tag_wakeup
- * target is the LCE that will receive a transfer for a transfer command
- * target_way_id is the way within the target LCE's set (computed from addr) to fill the data in to
+/**
+ *  bp_lce_cmd_cmd_s defines a command sent by a CCE to and LCE
+ *
+ *  src_id:        is the CCE sending the command
+ *  msg_type:      is the command
+ *  addr:          specifies the memory address associated with the command
+ *  way_id:        is the way within the set that addr maps to in the LCE that should be used for the command
+ *  state:         specifies the Coherence State to be used for invalidate, set_tag, and set_tag_wakeup
+ *  target:        is the LCE that will receive a transfer for a transfer command
+ *  target_way_id: is the way within the target LCE's set (computed from addr) to fill the data in to
  */
-`define declare_bp_cce_lce_cmd_s(num_cce_mp, num_lce_mp, addr_width_mp, lce_assoc_mp) \
+`define declare_bp_lce_cmd_cmd_s(num_cce_mp, num_lce_mp, addr_width_mp, lce_assoc_mp) \
   typedef struct packed                                         \
   {                                                             \
-    logic [`BSG_SAFE_CLOG2(num_lce_mp)-1:0]      dst_id;        \
-    logic [`BSG_SAFE_CLOG2(num_cce_mp)-1:0]      src_id;        \
-    bp_cce_lce_cmd_type_e                        msg_type;      \
-    logic [addr_width_mp-1:0]                    addr;          \
-    logic [`BSG_SAFE_CLOG2(lce_assoc_mp)-1:0]    way_id;        \
-    logic [`bp_cce_coh_bits-1:0]                 state;         \
-    logic [`BSG_SAFE_CLOG2(num_lce_mp)-1:0]      target;        \
     logic [`BSG_SAFE_CLOG2(lce_assoc_mp)-1:0]    target_way_id; \
-  }  bp_cce_lce_cmd_s
+    logic [`BSG_SAFE_CLOG2(num_lce_mp)-1:0]      target;        \
+    logic [`bp_coh_bits-1:0]                     state;         \
+    logic [addr_width_mp-1:0]                    addr;          \
+    bp_cce_lce_cmd_type_e                        msg_type;      \
+    logic [`BSG_SAFE_CLOG2(num_cce_mp)-1:0]      src_id;        \
+  }  bp_lce_cmd_cmd_s;                                          \
 
 
 /**
- *  bp_lce_data_cmd_s is used to send cache block data from CCE to LCE or from LCE to LCE.
+ *  bp_lce_cmd_data_s is used to send cache block data from CCE to LCE or from LCE to LCE.
  *
- *  dst_id:   LCE receiving the cache block data
- *  msg_type: indicates whether this is from CCE or from another LCE. LCE uses this field to decide
- *            what type of lce_resp to send. or whether this is non_cacheable data_cmd from CCE.
  *  way_id:   the way within the receiving LCE's target set to fill the data in to.
  *  data:     the cache block data
  */
+`define declare_bp_lce_cmd_data_s(data_width_mp) \
```

Comment:
I'd vote for just having a raw data field instead of a struct

Reply:
The Command network carries both data commands and regular commands. We could name the field "data" if we want, but the regular commands would still use those bits as the struct bp_lce_cmd_cmd_s above. We could also leave the field named as msg, and forget about the two data only structs, using the msg bits directly.

Reply:
Oh, I see. (Another thing that could be cleared up by a union?)

>  We could also leave the field named as msg, and forget about the two data only structs, using the msg bits directly.

Yeah, I think I like this better.